core/engine/src/builtins/date/utils.rs

```diff
@@ -750,22 +753,33 @@ pub(super) fn pad_six(t: u32, output: &mut [u8; 6]) -> JsStr<'_> {
 /// [spec-format]: https://tc39.es/ecma262/#sec-date-time-string-format
 pub(super) fn parse_date(date: &JsString, hooks: &dyn HostHooks) -> Option<i64> {
     // All characters must be ASCII so we can return early if we find a non-ASCII character.
-    let Ok(date) = date.to_std_string() else {
-        return None;
+    let owned_js_str = date.as_str();
+    let owned_string: String;
+    let date = match owned_js_str.variant() {
+        JsStrVariant::Latin1(s) =>
+        // SAFETY: Since all characters are ASCII we can safely convert this into str.
+        unsafe { str::from_utf8_unchecked(s) },
```

Comment:
This comment is incorrect though. `JsStrVariant::Latin1` could have latin1 encoded characters, which are incompatible with utf8.